eq-author-api/schema/resolvers/routing2/binaryExpression2/index.js

```diff
@@ -78,28 +105,239 @@ Resolvers.SelectedOptions2 = {
+          }
+        }, getOr([], "routing.rules", page))
+      ) {
+        return page;
+      }
+    }, pages);
+
+    const firstAnswer = first(getOr([], "answers", page));
+
+    const hasRoutableFirstAnswer =
+      firstAnswer &&
+      answerTypeToConditions.isAnswerTypeSupported(firstAnswer.type);
+    let condition;
+    if (hasRoutableFirstAnswer) {
+      condition = answerTypeToConditions.getDefault(firstAnswer.type);
+    }
+
+    const left = hasRoutableFirstAnswer
+      ? createLeftSide({
+          type: "Answer",
+          answerId: firstAnswer.id,
+        })
+      : createLeftSide({
+          type: "Null",
+          nullReason: "NoRoutableAnswerOnPage",
+        });
+
+    const expression = createExpression({
+      left,
+      condition: condition || "Equal",
+    });
+
+    expressionGroup.expressions.push(expression);
+
+    save(ctx.questionnaire);
+    return expression;
   },
-  updateBinaryExpression2: async (root, { input: { id, condition } }, ctx) =>
-    ctx.modifiers.BinaryExpression.update({ id, condition }),
+  updateBinaryExpression2: async (root, { input: { id, condition } }, ctx) => {
+    const pages = flatMap(section => section.pages, ctx.questionnaire.sections);
+    const rules = flatMap(
+      routing => getOr([], "rules", routing),
+      flatMap(page => page.routing, pages)
+    );
+
+    const expressionGroup = find(expressionGroup => {
+      if (some({ id }, expressionGroup.expressions)) {
+        return expressionGroup;
+      }
+    }, flatMap(rule => rule.expressionGroup, rules));
+
+    const expression = find({ id }, expressionGroup.expressions);
+
+    const leftSide = expression.left;
+
+    if (!leftSide) {
+      throw new Error("Can't have a condition without a left side");
+    }
+
+    const answers = flatMap(page => page.answers, pages);
+
+    const leftSideAnswer = find({ id: leftSide.answerId }, answers);
+    if (!answerTypeToConditions.isValid(leftSideAnswer.type, condition)) {
+      throw new Error(
+        "This condition is not compatible with the existing left side"
+      );
+    }
+
+    expression.condition = condition;
+
+    save(ctx.questionnaire);
+
+    return expression;
+  },
+
+  updateLeftSide2: (root, { input }, ctx) => {
+    const { expressionId, answerId } = input;
+
+    const pages = flatMap(section => section.pages, ctx.questionnaire.sections);
+    const rules = flatMap(
+      routing => getOr([], "rules", routing),
+      flatMap(page => page.routing, pages)
+    );
 
-  updateLeftSide2: (root, { input }, ctx) =>
-    ctx.modifiers.LeftSide.update(input),
+    const expressionGroup = find(expressionGroup => {
+      if (some({ id: input.id }, expressionGroup.expressions)) {
+        return expressionGroup;
+      }
+    }, flatMap(rule => rule.expressionGroup, rules));
+
+    const expression = find({ id: expressionId }, expressionGroup.expressions);
+
+    const answer = find({ id: answerId }, flatMap(page => page.answers, pages));
```

Comment:
This doesn't guarantee type or location in the document of the answer. So this might be wrong. I think this is a regression.

Reply:
It didn't seem to be validating this before?

https://github.com/ONSdigital/eq-author-app/blob/e58b720b9afb93d4745daec1ed3c05d34018739e/eq-author-api/modifiers/LeftSide/update.js

But you're right in that it probably should.

Reply:
Ignore this then as its not a regression. Happy to resolve this later.